File: spring-aws-s3-java/src/main/java/com/github/migangqui/spring/aws/s3/config/AmazonS3PropertiesConfig.java
package com.github.migangqui.spring.aws.s3.config;

import com.github.migangqui.spring.aws.s3.property.AmazonS3Properties;
import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Configuration;

@Configuration
@EnableConfigurationProperties(AmazonS3Properties.class)
public class AmazonS3PropertiesConfig {

}

File: spring-aws-s3-kotlin/src/main/kotlin/com/migangqui/spring/aws/s3/config/AmazonS3PropertiesConfig.kt
package com.migangqui.spring.aws.s3.config

import com.migangqui.spring.aws.s3.property.AmazonS3Properties
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Configuration

@Configuration
@EnableConfigurationProperties(AmazonS3Properties::class)
class AmazonS3PropertiesConfig